### ISSUE TITLE
Bug: `lines-around-directive` behaves differently when there is comment on the previous line

### ISSUE THREAD

**Issue by mvorisek** [eslint/eslint#16629]
### How to reproduce?

config: https://github.com/fomantic/Fomantic-UI/blob/f7259f27c8f041eb43997dcb12ee6b03984055ec/.eslintrc.js

code:

```js
(function ($, window, document) {
    'use strict';

    function isWindow(obj) {
        return obj != null && obj === obj.window;
    }
```

passes (no errors)

but

```
(function ($, window, document, undefined) { // eslint-disable-line no-shadow-restricted-names
    'use strict';

    function isWindow(obj) {
        return obj != null && obj === obj.window;
    }
```

is failing (and if a new line before `'use strict';` is added, there are another errors)

### What did you expect to happen?

no error, comment on the previous line should not change the eslint behaviour

### What actually happened?

https://github.com/fomantic/Fomantic-UI/actions/runs/3642985903/jobs/6150711682#step:5:11

```
/home/runner/work/Fomantic-UI/Fomantic-UI/src/definitions/behaviors/api.js
Error:   12:5  error  Expected newline before "use strict" directive  lines-around-directive
```

**Comment by mdjermanovic** [comment]
[lines-around-directive](https://eslint.org/docs/latest/rules/lines-around-directive) is deprecated, so we're not making any changes to it. You can use [padding-line-between-statements](https://eslint.org/docs/latest/rules/padding-line-between-statements) instead.

**Comment by mvorisek** [comment]
Thank you and sorry for this issue, I was convinced this has to be comming from the recommended eslint config as the rule was not in my config, but the problem is in Airbnb ruleset and I opened an issue there.
